### DISCUSSION TITLE
Rust embedded application

### DISCUSSION THREAD

**Discussion by expploitt** [WasmEdge/WasmEdge#1294] [Q&A]
Hello, I'm new to wasmEdge and I'm trying to understand the documentation and the Rust-SDK.

What I want is to have a Rust code that can execute multiples programs compiled in .wasm. I saw that the C API can import and execute a specific function but the Go API can load the .wasm and run the _start.

Is that possible in the C API? To load the wasm and the execute the functions _start with the 
```
WasmEdge_VMRunWasmFromFile(
    WasmEdge_VMContext *Cxt, const char *Path, const WasmEdge_String FuncName,
    const WasmEdge_Value *Params, const uint32_t ParamLen,
    WasmEdge_Value *Returns, const uint32_t ReturnLen)
```

function?

Thanks!!

**Comment by hydai** [comment]
Yes. Just like this example: https://github.com/WasmEdge/WasmEdge/blob/master/docs/book/en/src/embed/c.md#quick-start-guide-for-the-wasmedge-runner

And you will need to change the `FuncName` to `_start`. That's all.

**Comment by kmehant** [comment]
@hydai do we have support to expose go functions as wasm functions similar to using wasm-bindgen in rust?

**Reply by hydai** [reply]
Do you mean something like this?
https://github.com/second-state/WasmEdge-go-examples/blob/master/go_HostFunc/hostfunc.go

**Reply by kmehant** [reply]
No @hydai, I think the code is for using host functions in wasm, My question is regarding accessing functions defined in wasm from host language.

**Reply by hydai** [reply]
Sorry, do you want to call the wasm function from Golang? If so, here are examples: https://github.com/second-state/WasmEdge-go-examples/tree/master

**Reply by kmehant** [reply]
thanks, @hydai 

the examples use rust with wasm-bindgen, correct me if I am wrong. 

But my doubt was regarding using wasm modules compiled using lets say tinygo and with host lang as well in go. So basically wasm modules compiled from golang and then being used within go code to invoke some arbitrary function in wasm module.

When tried it, I see wasmedge runtime reporting the function is not found.

**Reply by hydai** [reply]
Could you provide a reproduce steps? AFAIK we can also use tinyGo to write applications, compile them into wasm, and call those functions from WasmEdge go bindings: https://github.com/second-state/WasmEdge-go-examples/tree/master/go_MemoryGreetTinyGo
